DriveClass/src/drive.cpp

```diff
@@ -0,0 +1,107 @@
+/*----------------------------------------------------------------------------*/
+/*                                                                            */
+/*    Module:       drive.cpp                                                  */
+/*    Author:       Alex Mei                                                  */
+/*    Created:      Sat Feb 22 2020                                           */
+/*    Description:  V5 project                                                */
+/*                                                                            */
+/*----------------------------------------------------------------------------*/
+
+// ---- START VEXCODE CONFIGURED DEVICES ----
+// ---- END VEXCODE CONFIGURED DEVICES ----
+
+#include "vex.h"
+
+using namespace vex;
+
+
+class Drive{
+  public:
+    static constexpr double PI = 3.14159265;
+    static constexpr double velocity = -1; //PLACEHOLDER TEST
+
+    //constructor
+    Drive(int portA = 0,int portB = 0){
+      delete LeftMotor;
+      delete RightMotor;
+      LeftMotor = new motor(portA, ratio18_1, false);
+      RightMotor = new motor(portB, ratio18_1, false);
+    }
+
+    //destructor
+    ~Drive(){
+      delete LeftMotor;
+      delete RightMotor;
+    }
+
+    void startMotors(){
+      LeftMotor->spin(forward);
+      RightMotor->spin(forward);
+    }
+
+    void move(double x, double y){
+      double unit_x = x / 100;
+      double unit_y = y / 100;
+      double unit_magnitude = sqrt(unit_x * unit_x + unit_y * unit_y);
+
+      //check domain
+      if(unit_magnitude > 1){
+        unit_magnitude = 1;
```

Comment:
What is this conditional statement for? I think unit_magnitude will never be over 1. x and y will always at most be 100.